src/analysisd/eventinfo.c

```diff
@@ -43,6 +42,9 @@ size_t field_offset[] = {
     offsetof(Eventinfo, location)
 };
 
+static pthread_mutex_t rule_level_mutex = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
You are not protecting all uses of rule level.
I think it would be better to remove this mutex for now and create a a new issue to fix all uses of rule level which are many more that these you are protecting.